database/sqlite/sqlite_aclk_node.c

```diff
@@ -35,15 +42,15 @@ void sql_build_node_info(struct aclk_database_worker_config *wc, struct aclk_dat
     node_info.data.kernel_name = host->system_info->kernel_name;
     node_info.data.kernel_version = host->system_info->kernel_version;
     node_info.data.architecture = host->system_info->architecture;
-    node_info.data.cpus = str2uint32_t(host->system_info->host_cores);
-    node_info.data.cpu_frequency = host->system_info->host_cpu_freq;
-    node_info.data.memory = host->system_info->host_ram_total;
-    node_info.data.disk_space = host->system_info->host_disk_space;
-    node_info.data.version = VERSION;
+    node_info.data.cpus = host->system_info->host_cores ? str2uint32_t(host->system_info->host_cores) : 0;
+    node_info.data.cpu_frequency = host->system_info->host_cpu_freq ? host->system_info->host_cpu_freq : "0";
+    node_info.data.memory = host->system_info->host_ram_total ? host->system_info->host_ram_total : "0";
+    node_info.data.disk_space = host->system_info->host_disk_space ? host->system_info->host_disk_space : "0";
+    node_info.data.version = host_version ? host_version : VERSION;
```

Comment:
is this correct? If `host!=localhost` and we can't figure the version from `host->receiver` we will fall back to `VERSION` which will be version of localhost. Shouldn't we do something like "unknown" in such case?

Reply:
I think the version of the sending child is there since the very first version of the streaming protocol. So, probably this is dead code. But I don't know the code base now, so I can't tell.

Reply:
> is this correct? If `host!=localhost` and we can't figure the version from `host->receiver` we will fall back to `VERSION` which will be version of localhost. Shouldn't we do something like "unknown" in such case?

yes, you are right, it is better to be "unknown" for now. An upcoming PR will properly resolve this

Reply:
> I think the version of the sending child is there since the very first version of the streaming protocol. So, probably this is dead code. But I don't know the code base now, so I can't tell.

yes it is parsed via "User-Agent:" but it will be only valid for a connected child (we need to store it properly to have the info for offline children as well)